### PR TITLE
feat: Added getInfo handler to return server metadata

### PR DESCRIPTION
**Pull request by JigarJoshi**

### CODE REVIEW

File: server/services/v1/api.go [hunk 1]
```diff
@@ -42,6 +43,8 @@ const (
 
 	documentPath        = collectionPath + "/documents"
 	documentPathPattern = documentPath + "/*"
+
+	infoPath = "/info"
```
Comment (himank) [inline server/services/v1/api.go:47]: but why not call it version?

Comment (himank) [reply server/services/v1/api.go:47]: what's the path look like?

Comment (JigarJoshi) [reply server/services/v1/api.go:47]: `/api/v1/info` 

we had a long discussion on this one here https://github.com/tigrisdata/tigris-api/pull/46#discussion_r862020814

File: Makefile [hunk 2]
```diff
@@ -20,7 +20,7 @@ all: server
 
 .PRECIOUS: ${PROTO_DIR}/%_openapi.yaml ${PROTO_DIR}/%.proto
 
-BUILD_PARAM=-tags=release -ldflags "-X 'main.Version=$(VERSION)' -X 'main.BuildHash=$(GIT_HASH)'" $(shell printenv BUILD_PARAM)
+BUILD_PARAM=-tags=release -ldflags "-X 'github.com/tigrisdata/tigris/util.Version=$(VERSION)' -X 'github.com/tigrisdata/tigris/util.BuildHash=$(GIT_HASH)'" $(shell printenv BUILD_PARAM)
```
Comment (efirs) [inline Makefile:23]: I've removed GIT_HASH in client repo, I think it's redundant. There is enough information in VERSION itself.
It'll be symbolic name if tagged and symbolic+hash if not tagged.

File: server/services/v1/api.go (hunk 1 above)
Comment (efirs) [reply server/services/v1/api.go:47]: I think it's good idea to have more generic `info` where we can put any other server related metadata.

File: Makefile (hunk 2 above)
Comment (JigarJoshi) [reply Makefile:23]: removed `BuildHash`

File: server/services/v1/api.go (hunk 1 above)
Comment (himank) [reply server/services/v1/api.go:47]: Makes sense, let's go with info.

**Review by himank — APPROVED** [review]

**Pull request merged**